src/holdup/cli.py

```diff
@@ -82,6 +89,32 @@ def __str__(self):
         return 'tcp://{0.host}:{0.port}'.format(self)
 
 
+class PgCheck(Check):
+    def __init__(self, host, port, database, user, password):
+        self.host = host
+        self.database = database
+        self.user = user
+        self.password = password
+        self.port = port
+
+    def run(self, options):
+        with closing(psycopg2.connect(host=self.host, database=self.database, user=self.user, password=self.password,
+                                      port=self.port, connect_timeout=options.check_timeout)) as conn:
+            # create a cursor
+            with closing(conn.cursor()) as cur:
+                start = time()
+                cur.execute('SELECT version()')
+                db_version = cur.fetchone()
+
+    def is_fatal_error(self, error):
+        representation = repr(error)
+        if 'password authentication failed' in representation or\
+            'database "{0}" does not exist'.format(self.database) in representation:
```

Comment:
Considering that db not existing is analogous to dns failing to resolve a hostname, and auth failures is analogous to permission denied errors when checking file/uds (the file/unix checks) then it would be inconsistent to have this distinction on failures.

It also makes code more complicated, lets revert it.